### PR TITLE
Support serialized `psp` model

### PR DESCRIPTION
**Pull request by simlmx**
## Description

* Load serialized models from the pv-site-prediction repo (added as a submodule)
* Run predictions on PV data from a database
* Support loading the netcdf NWP files from AWS
* Remove docker-compose
* Reorganize the code

## How Has This Been Tested?

* Added tests
* Ran the apply_model.py script in the dev environment on AWS

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ There are various different repos for ML research and development but this repo
 The main entry point is
 
 ```
-poetry run python pv_site_production/app.py
+poetry run python pv_site_production/scripts/apply_model.py
```
Comment (peterdudfield) [inline README.md:9]: should the main entry point be in scripts? I would have prefered it in `app.py` and then it can be tested. 

In different repos, Ive had scripts directory in the root folder, how come its with in `pv_site_production`

File: configuration_sample.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  cls: pv_site_production.data.nwp_data_sources.NwpDataSource
+  kwargs:
+    path: s3://nowcasting-nwp-development/data/latest.netcdf
+    storage_kwargs:
+      key: <aws key>
+      secret: <asw secret>
+
+pv_metadata_path: ./meta_inferred.csv
+
+pv_db_url: 'postgresql+psycopg2://<user>:<password>@<host>:<port>/<db_name>'
```
Comment (peterdudfield) [inline configuration_sample.yaml:16]: Ive done things before by setting ENV VARs in the configuration. This might be worth doing, as the nwp and pv data will be different for development and production envs

E.G. https://github.com/openclimatefix/nowcasting_forecast/blob/main/nowcasting_forecast/config/mvp_v2.yaml#L21

File: pv_site_production/data/nwp_data_sources.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        self._storage_kwargs = storage_kwargs
+        self._open()
+
+    def _open(self):
+        path = self._path
+
+        if path.endswith(".zarr"):
+            return super()._open()
+        elif path.endswith(".netcdf"):
+
```
Comment (peterdudfield) [inline pv_site_production/data/nwp_data_sources.py:34]: worth adding some logging? Perhaps this could be in a different PR

File: pv_site_production/data/pv_data_sources.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        # Add the metadata as coordinates to the pv_ids in the xr.Dataset.
+        da = da.assign_coords(
+            {
+                key: ("pv_id", [meta[pv_id][key] for pv_id in pv_ids])
+                for key in META_FILE_KEYS + META_DB_KEYS
+            }
+        )
+
+        if was_1d:
+            da = da.isel(pv_id=0)
```
Comment (peterdudfield) [inline pv_site_production/data/pv_data_sources.py:114]: could you add a comment to say what this is about?

File: pv_site_production/scripts/apply_model.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    # Wrap into a PV data source for the models.
+    _log.debug("Creating PV data source")
+    pv_data_source = DbPvDataSource(pv_db_connection, config["pv_metadata_path"])
+
+    _log.debug("Loading model")
+    model: PvSiteModel = get_model(config, pv_data_source)
+
+    pv_ids = pv_data_source.list_pv_ids()
+    # TODO Run for all the pv_ids
+    pv_ids = pv_ids[:10]
```
Comment (peterdudfield) [inline pv_site_production/scripts/apply_model.py:64]: could `n_pv_ids` be a clikc option? then we can set it to 10 for testing. Again, could move this to a seperate issue / PR

File: tests/models/cos/test_cos_model.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
+from datetime import datetime
```
Comment (peterdudfield) [inline tests/models/cos/test_cos_model.py:1]: Is it worth adding a test for the pv_nwp model? Sorry if i missed it

File: README.md (hunk 1 above)
Comment (simlmx) [reply README.md:9]: This is how I usually organize things, in case there are other entry points later, but I'm fine with reverting back to `app.py` at the root.

Just to be clear, which one do you prefer:
1. `/app.py`
2. `/pv_ste_production/app.py`
3. `/scripts/app.py`
?

Number 2. right?

File: configuration_sample.yaml (hunk 2 above)
Comment (simlmx) [reply configuration_sample.yaml:16]: Sure it can be environment variables. This is because you want to use the *same* config in both environments?

In this case:
```
 nwp_zarr_path: !ENV ${NWP_ZARR_PATH}
```

Where is `!ENV ${NWP_ZARR_PATH}` replaced with the actual value?

File: tests/models/cos/test_cos_model.py (hunk 6 above)
Comment (simlmx) [reply tests/models/cos/test_cos_model.py:1]: It's not but it's very simple (basically it just loads the serialized model), which the test on the script also does.
Happy to add one if you think it's worth it.

File: pv_site_production/data/nwp_data_sources.py (hunk 3 above)
Comment (simlmx) [reply pv_site_production/data/nwp_data_sources.py:34]: I'm not convinced we need logging in here, it's some pretty uneventful code!

File: README.md (hunk 1 above)
Comment (peterdudfield) [reply README.md]: number 2. please. But if there is a different way that is more common, then just say

File: configuration_sample.yaml (hunk 2 above)
Comment (peterdudfield) [reply configuration_sample.yaml:16]: so in the tests, you can set `NWP_ZARR_PATH` as a fixture, for example

in live, this can be set as an ENVVAR, and is set in `terraform` code.

Does that make sense?

File: pv_site_production/data/nwp_data_sources.py (hunk 3 above)
Comment (peterdudfield) [reply pv_site_production/data/nwp_data_sources.py:34]: ok, no worries

File: tests/models/cos/test_cos_model.py (hunk 6 above)
Comment (peterdudfield) [reply tests/models/cos/test_cos_model.py:1]: I thinks it would be worth it - thanks

File: README.md (hunk 1 above)
Comment (simlmx) [reply README.md]: I'm not sure what is more common. 2 it is, at least for now!

File: configuration_sample.yaml (hunk 2 above)
Comment (simlmx) [reply configuration_sample.yaml:16]: I understand the principle but I'm confused with the specifics in our case!
I did some work to make the `testcontainers` work and had to do something similar where I have placeholders in the config. It is similar to this but probably a slightly different implementation. Let's keep the discussion for that next PR if it's ok with you.

File: tests/models/cos/test_cos_model.py (hunk 6 above)
Comment (simlmx) [reply tests/models/cos/test_cos_model.py:1]: Added in 94ce3f50f0c60262b8b66274a296c015160e77dc
